streams/src/main/java/org/apache/kafka/streams/processor/internals/Task.java

```diff
@@ -154,6 +154,11 @@ public boolean isValidTransition(final State newState) {
      */
     void closeDirty();
 
+    /**
+     * Revive a closed task to a created one; should never throw an exception
+     */
+    void revive();
```

Comment:
Okay this might be a bit controversial: I tried to make the re-initialization logic as cheap as possible since otherwise we may be kicked out of the group for not calling consumer.poll in time. After looking at the source code I found the only part is the closure of the task-level sensors which I moved out of the `task.close` functions. So from CLOSED -> CREATED there's nothing necessary.